ci_framework/roles/reproducer/tasks/epel.yml

```diff
@@ -0,0 +1,39 @@
+---
+- name: Enable CRB repositories (CentOS)
+  when:
+    - ansible_facts['distribution'] == 'CentOS'
+  become: true
+  ansible.builtin.command:
+    cmd: dnf config-manager --set-enabled crb
+
+- name: Enable CRB repositories (RHEL)
+  when:
+    - ansible_facts['distribution'] == 'RedHat'
+  become: true
+  vars:
+    arch: "{{ ansible_facts['architecture'] }}"
+  ansible.builtin.command:
+    cmd: >-
+      subscription-manager repos
+      --enable codeready-builder-for-rhel-9-{{ arch }}-rpms
+
```

Comment:
It fails on RHEL machine if the system is not subscribed to subscription manager.
```
ASK [reproducer : Enable CRB repositories (RHEL) _raw_params=subscription-manager repos --enable codeready-builder-for-rhel-9-{{ arch }}-rpms] **************
Friday 20 October 2023  07:46:16 +0530 (0:00:00.023)       0:02:55.146 ******** 
fatal: [baremetal]: FAILED! => {"changed": true, "cmd": ["subscription-manager", "repos", "--enable", "codeready-builder-for-rhel-9-x86_64-rpms"], "delta": "0:00:00.243552", "end": "2023-10-20 05:16:21.881923", "msg": "non-zero return code", "rc": 1, "start": "2023-10-20 05:16:21.638371", "stderr": "", "stderr_lines": [], "stdout": "This system has no repositories available through subscriptions.", "stdout_lines": ["This system has no repositories available through subscriptions."]}
```
In CI, we never subscribe to subscription manager, we use the repos populated by rhos-release tool. If there is a missing repo, We add it.

In Devscript role we have a check regarding subscription manager. https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/devscripts/tasks/sub_tasks/11_rhsm.yml#L27

We can add the same here also or reuse the playbook.

The Devscripts role also provides option to pass custom crb repo: https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/devscripts/tasks/sub_tasks/12_yum_repo.yml#L21

I think we can use both of them here. What do you say?

Reply:
I'll wait to get access to the tasks files - Pragadee is modifying the devscripts role and will push those tasks elsewhere. I'd rather avoid duplicating code at this stage.